scripts/irods/test/test_symlink_operations.py

```diff
@@ -246,24 +246,28 @@ def test_irsync_iput_recursive_dangling_symlink_issues_3072_3988(self):
             # Iterate through each of the tests:
             ########
             for teststring in test_list:
+                with self.subTest(teststring):
+                    ignore_symlinks = '--link' in teststring
 
-                cmd = teststring.format(**locals())
-                stdout,stderr,_ = self.user0.run_icommand(cmd)
+                    cmd = teststring.format(**locals())
+                    stdout,stderr,_ = self.user0.run_icommand(cmd)
 
-                self.assertIn(recurse_fail_string, stdout,
-                              '{0}: Expected stdout: "...{1}...", got: "{2}"'.format(cmd, recurse_fail_string, stdout))
+                    if ignore_symlinks:
+                        self.assertEqual(str(), stdout, '{}: Expected stdout: "", got: "{}"'.format(cmd, stdout))
```

Comment:
Why is `str()` used instead of `""`? Is it triggering some conversion or something?

Reply:
No real reason. I have updated all of these to just use `''`'.